src/addon/validation.cpp

```diff
@@ -94,6 +104,11 @@ bool addon_filename_legal(const std::string& name)
 	   name.size() > 255) {
 		return false;
 	} else {
+		const std::string stem = boost::algorithm::to_upper_copy(name.substr(0, name.find('.')), std::locale::classic());
```

Comment:
Please use [`filesystem::base_name()`](https://github.com/wesnoth/wesnoth/blob/4f225f56e0acd6b059dc5cbba20e85c6a905714c/src/filesystem.hpp#L190) to extract the filename instead.

Reply:
That's not going to give the expected results. For example, `con.test.test` is handled by Windows as if referring to `CON`, but `filesystem::basename()` is going to return `con.test` instead and therefore not match the list.

Reply:
I see. Please add a comment to mention that, then.